app/src/main/java/swen_anigans/mathematicfanatic/ClassroomActivity.java

```diff
@@ -168,21 +150,18 @@ public void goToLocker(View view){
     }
 
     public void checkAnswer(View view){
-        EditText answerInput = (EditText) findViewById(R.id.editClassroomAnswer);
-        String answer = answerInput.getText().toString();
-        CharSequence text;
-        if (!answer.isEmpty()) {
-            int quizAnswer = Integer.parseInt(answer);
-            answers.set(pageNumber-1, quizAnswer);
-            boolean correct;
-            if(quizAnswer == expectedAnswers.get(pageNumber - 1)){
-                text = "Correct!";
+        TextView answerInput = (TextView) findViewById(R.id.editClassroomAnswer);
+        String answerText = answerInput.getText().toString();
+        String text;
+        if(!answerText.isEmpty()){
```

Comment:
I think it would make sense to just do save answer, then do something like the above if. That way we know that we are saving the answer that was inputted before we try to do anything. I think it would make it easier to read and obvious what check answer is actually trying to do.